contracts/SpokePool.sol

```diff
@@ -845,7 +841,11 @@ abstract contract SpokePool is
     {
         // Exclusivity deadline is inclusive and is the latest timestamp that the exclusive relayer has sole right
         // to fill the relay.
-        if (relayData.exclusivityDeadline >= getCurrentTime() && relayData.exclusiveRelayer != msg.sender) {
+        if (
+            relayData.exclusiveRelayer != address(0) &&
+            relayData.exclusivityDeadline >= getCurrentTime() &&
+            relayData.exclusiveRelayer != msg.sender
```

Comment:
nit: can we move this to the first check. Small gas optimization.

We think exclusivity will be the most common case, so we probably expect that most fills will be able to exit this if on the `relayData.exclusiveRelayer != msg.sender` check.